File: Source/WickedSick.Fayde.Client.Engine/Javascript/Media/Animation/ColorAnimationUsingKeyFrames.js
/// <reference path="AnimationUsingKeyFrames.js"/>
/// <reference path="ColorKeyFrame.js"/>
/// CODE

(function (namespace) {
    var ColorAnimationUsingKeyFrames = Nullstone.Create("ColorAnimationUsingKeyFrames", namespace.AnimationUsingKeyFrames);

    //#region Properties

    ColorAnimationUsingKeyFrames.KeyFramesProperty = DependencyProperty.RegisterFull("KeyFrames", function () { return namespace.ColorKeyFrameCollection; }, ColorAnimationUsingKeyFrames, undefined, undefined, { GetValue: function () { return new namespace.ColorKeyFrameCollection(); } });

    Nullstone.AutoProperties(ColorAnimationUsingKeyFrames, [
        ColorAnimationUsingKeyFrames.KeyFramesProperty
    ]);

    //#endregion

    //#region Annotations

    ColorAnimationUsingKeyFrames.Annotations = {
        ContentProperty: ColorAnimationUsingKeyFrames.KeyFramesProperty
    };

    //#endregion

    namespace.ColorAnimationUsingKeyFrames = Nullstone.FinishCreate(ColorAnimationUsingKeyFrames);
})(Nullstone.Namespace("Fayde.Media.Animation"));

File: Source/Fayde.Server/Media/Animation/ColorAnimationUsingKeyFrames.cs
using Fayde.Core;
using Fayde.Xaml.Metadata;

namespace Fayde.Media.Animation
{
    public class ColorAnimationUsingKeyFrames : Timeline
    {
        public static readonly PropertyDescription KeyFramesProperty = PropertyDescription.Register("KeyFrames", typeof(DependencyObjectCollection<ColorKeyFrame>), typeof(ColorAnimationUsingKeyFrames), true);
    }
}